java/src/main/java/com/xebia/functional/xef/java/auto/AIScope.java

```diff
@@ -41,31 +38,26 @@ public class AIScope implements AutoCloseable {
     private final CoreAIScope scope;
     private final ObjectMapper om;
     private final JsonSchemaGenerator schemaGen;
-    private final KtorOpenAIClient client;
+    private final AIClient client;
     private final ExecutorService executorService;
     private final CoroutineScope coroutineScope;
 
-    public AIScope(ObjectMapper om, OpenAIConfig config, ExecutorService executorService) {
+    public AIScope(ObjectMapper om, AIRuntime runtime, ExecutorService executorService) {
         this.om = om;
         this.executorService = executorService;
         this.coroutineScope = () -> ExecutorsKt.from(executorService).plus(JobKt.Job(null));
         this.schemaGen = new JsonSchemaGenerator(om);
-        this.client = new KtorOpenAIClient(config);
-        Embeddings embeddings = new OpenAIEmbeddings(config, client);
-        VectorStore vectorStore = new LocalVectorStore(embeddings);
-        this.scope = new CoreAIScope(LLMModel.getGPT_3_5_TURBO(), LLMModel.getGPT_3_5_TURBO_FUNCTIONS(), client, vectorStore, embeddings, 3, "user", false, 0.4, 1, 20, 500);
-    }
-
-    public AIScope(OpenAIConfig config) {
-        this(new ObjectMapper(), config, Executors.newCachedThreadPool(new AIScopeThreadFactory()));
+        this.client = runtime.getClient();
+        VectorStore vectorStore = new LocalVectorStore(runtime.getEmbeddings());
+        this.scope = new CoreAIScope(LLMModel.getGPT_3_5_TURBO(), LLMModel.getGPT_3_5_TURBO_FUNCTIONS(), client, vectorStore, runtime.getEmbeddings(), 3, "user", false, 0.4, 1, 20, 500);
     }
 
-    public AIScope(OpenAIConfig config, ExecutorService executorService) {
-        this(new ObjectMapper(), config, executorService);
+    public AIScope(AIRuntime runtime, ExecutorService executorService) {
+        this(new ObjectMapper(), runtime, Executors.newCachedThreadPool(new AIScopeThreadFactory()));
```

Comment:
Overriding user provided `ExecutorService`.

```suggestion
    public AIScope(AIRuntime runtime, ExecutorService executorService) {
        this(new ObjectMapper(), runtime, executorService);
```